.emacs.d/lisp/visual/text-visuals.el

```diff
@@ -29,10 +34,35 @@
   (interactive "nNew font height: ")
   (set-face-attribute 'default nil :height (floor (* 10.0 height))))
 
-;; Set the font height to the default height.
-(kotct/font-set-height kotct/font-default-height)
+(defvar kotct/font-rescale-factor
+  1.0
+  "The proportion that the font height change is multiplied by.")
+
+(defun kotct/font-rescale-height (amount)
+  "Sets the `:height' face attribute on the `default' face to
+`kotct/font-rescale-factor' * AMOUNT + the current height."
+  (let ((new-size (+ (* kotct/font-rescale-factor amount)
+                     (/ (face-attribute 'default :height) 10.0))))
+    (set-face-attribute 'default nil :height (floor (* 10.0 new-size)))))
+
+(defun kotct/font-upscale-height () (interactive "*") (kotct/font-rescale-height +1.0))
+(defun kotct/font-downscale-height () (interactive "*") (kotct/font-rescale-height -1.0))
+(defun kotct/font-reset-height () (interactive "*") (kotct/font-set-height kotct/font-default-height))
+
+(global-unset-key (kbd "C-x C-+"))
+(global-unset-key (kbd "C-x C-="))
+(global-unset-key (kbd "C-x C--"))
+(global-unset-key (kbd "C-x C-0"))
```

Comment:
They probably aren't, to be honest. I'll rework this later this evening after I'm done working through some of my schoolwork.